### PR TITLE
Add HTTP status code 103

### PR DESCRIPTION
**Pull request by Philippus**

### CODE REVIEW

**Comment by ktoso** [comment]
(make sure has a ticket or the PR is assigned to the milestone)

**Comment by jrudolph** [comment]
Looks good, I wonder if we should mark it as `@ApiMayChange` for now since the RFC is still experimental?

**Comment by ktoso** [comment]
Good catch, yeah lets. Would you add that annotation please @Philippus ?

**Comment by Philippus** [comment]
the RFC is experimental but the HTTP status code is already in the official registry: 
https://www.iana.org/assignments/http-status-codes/http-status-codes.xhtml
But happy to add it if you want of course.

**Comment by jlprat** [comment]
We can remove the `@ApiMayChange` at any time :)

File: akka-http-core/src/main/scala/akka/http/scaladsl/model/StatusCode.scala [hunk 1]
```diff
@@ -104,6 +105,9 @@ object StatusCodes extends ObjectRegistry[Int, StatusCode] {
   val SwitchingProtocols = reg(i(101)("Switching Protocols", "The server is switching protocols, because the client requested the switch."))
   val Processing         = reg(i(102)("Processing", "The server is processing the request, but no response is available yet."))
 
+  @ApiMayChange
+  val EarlyHints         = reg(i(103)("Early Hints", "The server is likely to send a final response with the header fields included in the response.")) // RFC 8297
```
Comment (raboof) [inline akka-http-core/src/main/scala/akka/http/scaladsl/model/StatusCode.scala]: The text is a bit puzzling to me, but that's how the RFC also describes it, so OK :).

File: akka-http-core/src/main/java/akka/http/javadsl/model/StatusCodes.java [hunk 2]
```diff
@@ -19,6 +19,7 @@ private StatusCodes() {}
     public static final StatusCode CONTINUE = akka.http.scaladsl.model.StatusCodes.Continue();
     public static final StatusCode SWITCHING_PROTOCOLS = akka.http.scaladsl.model.StatusCodes.SwitchingProtocols();
     public static final StatusCode PROCESSING = akka.http.scaladsl.model.StatusCodes.Processing();
+    public static final StatusCode EARLY_HINTS = akka.http.scaladsl.model.StatusCodes.EarlyHints();
```
Comment (jlprat) [inline akka-http-core/src/main/java/akka/http/javadsl/model/StatusCodes.java]: I guess `@ApiMayChange` also needs to go here :)

**Comment by jrudolph** [comment]
unrelated failure

PLS BUILD

**Comment by jrudolph** [comment]
Thanks, @Philippus!

**Pull request merged**